RecoMTD/TrackExtender/plugins/TrackExtenderWithMTD.cc

```diff
@@ -63,102 +63,175 @@ using namespace edm;
 using namespace reco;
 
 namespace {
+  constexpr float c_cm_ns = geant_units::operators::convertMmToCm(CLHEP::c_light);  // [mm/ns] -> [cm/ns]
+  constexpr float c_inv = 1.0f / c_cm_ns;
+
   class MTDHitMatchingInfo {
   public:
     MTDHitMatchingInfo() {
       hit = nullptr;
-      estChi2 = std::numeric_limits<double>::max();
-      timeChi2 = std::numeric_limits<double>::max();
+      estChi2 = std::numeric_limits<float>::max();
+      timeChi2 = std::numeric_limits<float>::max();
     }
 
     //Operator used to sort the hits while performing the matching step at the MTD
     inline bool operator<(const MTDHitMatchingInfo& m2) const {
       //only for good matching in time use estChi2, otherwise use mostly time compatibility
-      constexpr double chi2_cut = 10.;
-      constexpr double low_weight = 3.;
-      constexpr double high_weight = 8.;
+      constexpr float chi2_cut = 10.f;
+      constexpr float low_weight = 3.f;
+      constexpr float high_weight = 8.f;
       if (timeChi2 < chi2_cut && m2.timeChi2 < chi2_cut)
         return chi2(low_weight) < m2.chi2(low_weight);
       else
         return chi2(high_weight) < m2.chi2(high_weight);
     }
 
-    inline double chi2(double timeWeight = 1.) const { return estChi2 + timeWeight * timeChi2; }
+    inline float chi2(float timeWeight = 1.f) const { return estChi2 + timeWeight * timeChi2; }
 
     const MTDTrackingRecHit* hit;
-    double estChi2;
-    double timeChi2;
+    float estChi2;
+    float timeChi2;
+  };
+
+  class TrackSegments {
+  public:
+    TrackSegments() = default;
+
+    inline uint32_t addSegment(float tPath, float tMom2) {
+      segmentPathOvc_.emplace_back(static_cast<float>(tPath * c_inv));
+      segmentMom2_.emplace_back(static_cast<float>(tMom2));
```

Comment:
```suggestion
      segmentPathOvc_.emplace_back(tPath * c_inv);
      segmentMom2_.emplace_back(tMom2);
```
everything is in float now

Reply:
ok